clippy_lints/src/methods/mod.rs

```diff
@@ -2225,8 +2225,27 @@ declare_clippy_lint! {
 }
 
 declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for calling `take` function after `as_ref`.
     ///
     /// ### Why is this bad?
+    /// Redundant code.
+    /// ```ignore
+    /// let x = Some(2);
+    /// let y = x.as_ref().take();
+    /// ```
```

Comment:
This section should conceptually explain why this is bad, rather than with examples. If the examples are not good enough, please add better examples there. But I would prefer to keep this section free of code examples for all lints (there are always exceptions of course).

Reply:
@flip1995 updated